src/python/zquantum/core/measurement.py

```diff
@@ -392,3 +393,96 @@ def convert_bitstring_to_int(bitstring: Iterable[int]) -> int:
             significant (little endian).
     """
     return int("".join(str(bit) for bit in bitstring[::-1]), 2)
+
+class Measurements:
+    """ A class representing measurements from a quantum circuit. The bitstrings variable represents the internal
+    data structure of the Measurements class. It is expressed as a list of tuples wherein each tuple is a measurement
+    and the value of the tuple at a given index is the measured bit-value of the qubit (indexed from N-1 -> 0) """
+
+    def __init__(self, bitstrings: Optional[List[Tuple[int]]] = None):
+        if bitstrings is None:
+            self.bitstrings = []
+        else:
+            self.bitstrings = bitstrings
+
+    @classmethod
+    def from_counts(cls, counts: Dict):
+        """ Create an instance of the Measurements class from a dictionary
+        
+        Args:
+            counts (dict): mapping of bitstrings to integers representing the number of times the bitstring was measured
+        """
+        measurements = cls()
+        measurements.add_counts(counts)
+        return measurements
+
+    @classmethod
+    def load_from_file(cls, file: TextIO):
+        """ Load a set of measurements from file 
+        
+        Args:
+            file (str or file-like object): the name of the file, or a file-like object
+        """        
+        if isinstance(file, str):
+            with open(file, 'r') as f:
+                data = json.load(f)
+        else:
+            data = json.load(file)
+
+        bitstrings = []
+        for bitstring in data["bitstrings"]:
+            bitstrings.append(tuple(bitstring))
+
+        return cls(bitstrings=bitstrings)
+
+    def save(self, filename: String):
```

Comment:
I think it should be `str` not `String` – the typing requires those funny capital letter names only for the "special type" or "collection type" of variables.
See: https://docs.python.org/3/library/typing.html

Reply:
Addressed here: 
7288dca